tools/ctl/ctl.go

```diff
@@ -216,6 +217,94 @@ var getResults = &cobra.Command{
 	},
 }
 
+var missingAttestations = &cobra.Command{
+	Use:   "missing -project <ID> -run <ID> -attestation-bucket <name> -format <list|bench>",
+	Short: "Find successes from a smoketest run that don't exist in the attestation bucket.",
+	Args:  cobra.NoArgs,
+	Run: func(cmd *cobra.Command, args []string) {
+		log.Default().SetOutput(cmd.ErrOrStderr())
+		var passing []firestore.Rebuild
+		{
+			req, err := buildFetchRebuildRequest(cmd.Context(), "", *runFlag, "", *clean)
+			if err != nil {
+				log.Fatal(err)
+			}
+			fireClient, err := firestore.NewClient(cmd.Context(), *project)
+			if err != nil {
+				log.Fatal(err)
+			}
+			rebuilds, err := fireClient.FetchRebuilds(cmd.Context(), req)
+			if err != nil {
+				log.Fatal(err)
+			}
+			for _, r := range rebuilds {
+				if r.Success {
+					passing = append(passing, r)
+				}
+			}
+		}
+		var missing []firestore.Rebuild
+		{
+			ctx := cmd.Context()
+			ctx = context.WithValue(ctx, rebuild.RunID, "")
+			ctx = context.WithValue(ctx, rebuild.GCSClientOptionsID, []option.ClientOption{option.WithoutAuthentication()})
+			attestation, err := rebuild.NewGCSStore(ctx, "gs://"+*attestationBucket)
+			if err != nil {
+				log.Fatal(errors.Wrapf(err, "creating attestation store"))
+			}
+			log.Println("Checking all successful packages for attestation...")
+			bar := pb.New(len(passing))
+			bar.ShowTimeLeft = true
+			bar.Output = cmd.ErrOrStderr()
+			bar.Start()
+			for _, rb := range passing {
+				r, _, err := attestation.Reader(ctx, rebuild.Asset{Target: rb.Target(), Type: rebuild.AttestationBundleAsset})
+				if errors.Is(err, rebuild.ErrAssetNotFound) {
+					missing = append(missing, rb)
+				} else if err != nil {
+					log.Fatal(errors.Wrapf(err, "failed attempting to read %v", rb.Target()))
+				} else {
+					defer r.Close()
+				}
+				bar.Increment()
+			}
+			bar.Finish()
+		}
+		slices.SortFunc(missing, func(a firestore.Rebuild, b firestore.Rebuild) int { return strings.Compare(a.ID(), b.ID()) })
+		switch *format {
+		case "list":
+			for _, rb := range missing {
+				fmt.Println(rb.Target())
+			}
+		case "bench":
+			var ps benchmark.PackageSet
+			ps.Count = len(missing)
+			for _, r := range missing {
+				idx := -1
+				for i, psp := range ps.Packages {
+					if psp.Name == r.Package {
+						idx = i
+						break
+					}
+				}
+				if idx == -1 {
+					ps.Packages = append(ps.Packages, benchmark.Package{Name: r.Package, Ecosystem: r.Ecosystem})
+					idx = len(ps.Packages) - 1
+				}
+				ps.Packages[idx].Versions = append(ps.Packages[idx].Versions, r.Version)
+			}
+			ps.Updated = time.Now()
+			b, err := json.MarshalIndent(ps, "", "  ")
+			if err != nil {
+				log.Fatal(errors.Wrap(err, "marshalling benchmark"))
+			}
+			fmt.Println(string(b))
```

Comment:
use the cmd output

Reply:
Copied from get-results, but I can change it.